doc/building.md

```diff
@@ -1,4 +1,60 @@
-Building nyan
-=============
+# Building nyan
+
+This project requires a C++14 compiler (e.g gcc >= 5.1 or clang >= 3.4) and uses Flex, make, automake, autoconf and libtool (any "recent" version should do).
+
+## Dependencies install
+
+###### Ubuntu 15.10 
+
+Required packages :
+```
+sudo apt-get update
+sudo apt-get install make libtool autoconf automake flex bison git
+```
+
+For gcc : `sudo apt-get install gcc-5 g++-5`
+
+For clang : `sudo apt-get install clang-3.5 clang++-3.5`
```

Comment:
I see your point and it seems legit to me. Unfortunately, I really don't know why everything seems to be excessively complicated when it comes to gcc and clang packages on Ubuntu.

Important note: Here, for gcc and clang on Ubuntu, the version numbers is a part of the package name (just like python2, python3, etc...). You have thus the ability to install multiple versions of the same software from different package (e.g:  `foo` from `foo-1` and/or `foo-2`). The `update-alternatives` utility software is here to help you configure the current default `alternative`.

To answer your question, here is what happens on ubuntu 15.10 when you try to install clang++ : 
```
# apt-get update && apt-get install clang++
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 python-clang-3.4 : Breaks: python-clang-3.5 but 1:3.5.2-2 is to be installed
 python-clang-3.5 : Breaks: python-clang-3.4 but 1:3.4.2-13ubuntu2.1 is to be installed
                    Breaks: python-clang-3.6 but 1:3.6.2-1 is to be installed
 python-clang-3.6 : Breaks: python-clang-3.4 but 1:3.4.2-13ubuntu2.1 is to be installed
 python-clang-3.7 : Breaks: python-clang-3.4 but 1:3.4.2-13ubuntu2.1 is to be installed
                    Breaks: python-clang-3.5 but 1:3.5.2-2 is to be installed
                    Breaks: python-clang-3.6 but 1:3.6.2-1 is to be installed
```
I just can't figure out what's happening here :(

However, the following does the job for gcc-5 and g++-5 :
```# apt-get update && apt-get install gcc g++```
:)

Conclusion IMHO, on Ubuntu 15.10 (and probably 16.04) we *need* to specify the clang version (versions 3.4, 3.5, 3.6 and 3.7 are currently available). The best I can do is removing the gcc version and upgrading clang to 3.7. What do you think ?